esp-hal/src/gpio/mod.rs

```diff
@@ -1044,57 +1033,36 @@ fn on_pin_irq(pin_nr: u8) {
     asynch::PIN_WAKERS[pin_nr as usize].wake(); // wake task
 }
 
-#[doc(hidden)]
-pub trait GpioProperties {
-    type Bank: BankGpioRegisterAccess;
-    type InterruptStatus: InterruptStatusRegisterAccess;
-
-    type IsOutput: BooleanType;
-    type IsAnalog: BooleanType;
-    type IsTouch: BooleanType;
-
-    fn degrade_pin(&self, _: private::Internal) -> AnyPin;
-    fn output_signals() -> [Option<OutputSignal>; 6];
-    fn input_signals() -> [Option<InputSignal>; 6];
-}
-
 #[doc(hidden)]
 #[macro_export]
 macro_rules! if_output_pin {
-    (InputOnlyAnalog, { $($then:tt)* } else { $($else:tt)* } ) => { $($else)* };
     (InputOutputAnalog, { $($then:tt)* } else { $($else:tt)* } ) => { $($then)* };
     (InputOutputAnalogTouch, { $($then:tt)* } else { $($else:tt)* } ) => { $($then)* };
     (InputOutput, { $($then:tt)* } else { $($else:tt)* } ) => { $($then)* };
+    ($other:ident, { $($then:tt)* } else { $($else:tt)* } ) => { $($else)* };
```

Comment:
This macro didn't handle `InputOnly` before, even though we don't have any such pin.